### ISSUE TITLE
add indentation to button text on Windows

### ISSUE THREAD

**Issue by veksha** [crow-translate/crow-translate#531]
**Describe the problem or limitation you are having**

On Windows there is no space (indent) between button text and button edges.

![brave_U5jDM79O0r](https://user-images.githubusercontent.com/275333/232466706-578ab3af-aa02-44f2-8ec6-f6f0ecefc83c.png)
![brave_yWxgO2x3dO](https://user-images.githubusercontent.com/275333/232466712-d8c3cf18-b9e7-440c-a4c7-a65e60e4a51e.png)

**Describe the solution you'd like**
add indentation to button text on Windows.

**Comment by Shatur** [comment]
I think this could work, but I would rather rework the widget. Currently on the right it's the same widget as on the left, but with "right-to-left" enabled. We probably shouldn't do it.

**Comment by veksha** [comment]
I don't understand what you mean by "rework the widget". Will it help with padding?

**Comment by Shatur** [comment]
Yes, this is why I suggesting it. Buttons are standard, but they located in custom widget: https://github.com/crow-translate/crow-translate/blob/master/src/languagebuttonswidget.cpp
It's not the widget that causes the problem, but how Qt displays buttons on Windows. But if we stop using "right-to-left" for right buttons, then this will solve the problem.

**Comment by veksha** [comment]
for these buttons too?

![brave_EAhI5CuPBt](https://user-images.githubusercontent.com/275333/232516847-15920747-1473-414d-bfa0-3ff49530bab8.png)
![brave_fZqciKhMMd](https://user-images.githubusercontent.com/275333/232516858-eb21ceb4-18db-4a90-b829-f24d29ffc26e.png)

**Comment by Shatur** [comment]
No, these buttons do not use right-to-left layout. If I understand correctly, problem only with these buttons:
![изображение](https://user-images.githubusercontent.com/22453358/232528124-3db234b0-f578-4231-b16e-7538e69ef96f.png)
The buttons on the left and right are inside the `LanguageButtonsWidget` widget which contains the language switching logic. But the version on the right is different because it uses the `QLayout`'s right-to-left parameter. I did this to get a "mirror" look. But it seems to have a nasty effect on Windows. Therefore, I think that we need to remake this widget so that it can do a "mirror view", but without using the above parameter.

**Comment by veksha** [comment]
sorry for confusion.
no. the problem is that **every** button on Windows has no padding. padding on every side (left/right/top/bottom). and therefore button sizes are smaller.

**Comment by Shatur** [comment]
Oh, I see! We could "hack" the style, but I would rather not to. It's the default style of buttons on Windows, I didn't change it.
Qt support styling, so you could try to override the default style, but setting the `QT_STYLE_OVERRIDE`. But it will require searching for DLLs and put them into `plugins/styles`.

**Comment by veksha** [comment]
found simple solution:
```
You can set the padding for all buttons in a window/widget with:

QPushButton { padding: 10px; }
Much better than applying to each subwidget.
```

but if you don't want to mess with it, i understand. you can close the issue then.

**Comment by Shatur** [comment]
> but if you don't want to mess with it, i understand. you can close the issue then.

Yes, I would prefer to avoid applying custom styling for different OSes.

**Comment by veksha** [comment]
how about this?

https://user-images.githubusercontent.com/275333/232743404-d5ff55a0-76aa-42df-997b-063ec0f18050.mp4



```diff
diff --git a/src/languagebuttonswidget.cpp b/src/languagebuttonswidget.cpp
index 874ba44..5e5eeb4 100644
--- a/src/languagebuttonswidget.cpp
+++ b/src/languagebuttonswidget.cpp
@@ -513,9 +513,59 @@ void LanguageButtonsWidget::addOrCheckLanguage(QOnlineTranslator::Language lang)
     m_buttonGroup->buttons().constLast()->setChecked(true);
 }
 
+#include <qstylepainter.h>
+#include <qstyleoption.h>
+class CustomToolButton : public QToolButton
+{
+public:
+    explicit CustomToolButton(QWidget *parent = nullptr)
+        : QToolButton(parent)
+    {
+        // Set the initial icon size
+        int iconSize = 24;
+        setIconSize(QSize(iconSize, iconSize));
+    }
+
+protected:
+    void paintEvent(QPaintEvent *event) override
+    {
+        if (icon().isNull()) {
+            QToolButton::paintEvent(event);
+            return;
+        }
+        
+        QPainter painter(this);
+        QStyleOptionToolButton option;
+        initStyleOption(&option);
+    
+        auto buttonRect = option.rect;
+        buttonRect.adjust(7, 7, -7, -7);
+    
+        auto oldText = option.text;
+        option.text = ""; // make it invisible for drawComplexControl()
+        option.iconSize = QSize(0, 0); // make it invisible for drawComplexControl()
+        
+        // drawing naked button
+        style()->drawComplexControl(QStyle::CC_ToolButton, &option, &painter, this);
+        option.text = oldText; // restore text
+        
+        // drawing text
+        if(isChecked())buttonRect = buttonRect.translated(1, 1);
+        auto align = layoutDirection() == Qt::LeftToRight ? Qt::AlignRight : Qt::AlignLeft;
+        style()->drawItemText(&painter, buttonRect, align | Qt::AlignVCenter, palette(), true, option.text);
+        
+        // drawing icon
+        align = layoutDirection() == Qt::LeftToRight ? Qt::AlignLeft : Qt::AlignRight;
+        style()->drawItemPixmap(&painter, buttonRect, align, option.icon.pixmap(
+            iconSize().scaled(buttonRect.width(), buttonRect.height(), Qt::KeepAspectRatio)
+        ));
+    }
+};
+
 void LanguageButtonsWidget::addButton(QOnlineTranslator::Language lang)
 {
-    auto *button = new QToolButton;
+    auto *button = new CustomToolButton;
+    //auto *button = new QToolButton;
     button->setCheckable(true);
     button->setToolButtonStyle(Qt::ToolButtonTextBesideIcon);
     button->setSizePolicy(QSizePolicy::Preferred, QSizePolicy::Preferred); // To make the same size for all buttons (without it "Auto" button can look different)

```

**Comment by Shatur** [comment]
Sorry, but same here: I prefer to not style buttons. Because custom styling may look good with Windows 10, but can look bad with different Linux distros or Windows 11. If the buttons doesn't look nice by default, well, it's Windows problem.

**Comment by veksha** [comment]
ok. i think it's qt problem.

**Comment by Shatur** [comment]
> ok. i think it's qt problem.

That's also a possibility, yes. Qt draws buttons on its own, but mimics the native interface as much as possible. So if there is a visual difference between using WinAPI and Qt - it's a Qt issue. You can check other apps that use WinAPI, like foobar2000.
